### PR TITLE
Code Refactoring | components report

### PR DESCRIPTION
**Pull request by vishnu-aurx**

### CODE REVIEW

File: core/src/main/java/com/aurx/core/utils/ComponentUtils.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
+package com.aurx.core.utils;
+
+public class ComponentUtils {
```
Comment (abhinav-aggarwal-aem) [inline core/src/main/java/com/aurx/core/utils/ComponentUtils.java]: is this a util?

**Review by abhinav-aggarwal-aem — DISMISSED** [review]

File: core/src/main/java/com/aurx/core/models/ComponentReportModel.java [hunk 2]
```diff
@@ -17,12 +17,13 @@ public class ComponentReportModel {
   @OSGiService
   private ComponentReportService componentReportService;
    private List<ComponentReport> comopnentHierachyList;
-  private final Logger logger= LoggerFactory.getLogger(ComponentReportModel.class);
+  private static  final Logger logger= LoggerFactory.getLogger(ComponentReportModel.class);
```
Comment (akashsen-aurx) [inline core/src/main/java/com/aurx/core/models/ComponentReportModel.java:33]: java doc

File: core/src/main/java/com/aurx/core/models/ComponentReportModel.java [hunk 3]
```diff
@@ -17,12 +17,13 @@ public class ComponentReportModel {
...
   @PostConstruct
   protected void init(){
     comopnentHierachyList = new ArrayList<>();
-    logger.info("inside the init method");
+    logger.info("============inside the init method===========");
```
Comment (akashsen-aurx) [inline core/src/main/java/com/aurx/core/models/ComponentReportModel.java]: please remove this unnecessary symbol

File: core/src/main/java/com/aurx/core/pojo/ComponentReport.java [hunk 4]
```diff
@@ -3,13 +3,15 @@
 import java.util.Comparator;
 
 public class ComponentReport implements Comparator<ComponentReport> {
+
   private String pagePath;
   private String page;
- private String title;
- private String group;
- private String pathOfComponent;
+  private String title;
```
Comment (akashsen-aurx) [inline core/src/main/java/com/aurx/core/pojo/ComponentReport.java:21]: add java doc of every variable

File: core/src/main/java/com/aurx/core/pojo/ComponentReport.java [hunk 5]
```diff
@@ -70,10 +72,11 @@ public String toString() {
 
   @Override
   public int compare(ComponentReport componentReport1, ComponentReport componentReport2) {
-    int compareGroup=componentReport1.group.compareToIgnoreCase(componentReport2.group);
-    if(compareGroup ==0){
-      return  componentReport1.getTitle().compareToIgnoreCase(componentReport2.getTitle());
-    }else
-    return compareGroup ;
+    int compareGroup = componentReport1.group.compareToIgnoreCase(componentReport2.group);
```
Comment (akashsen-aurx) [inline core/src/main/java/com/aurx/core/pojo/ComponentReport.java:148]: add java doc

File: core/src/main/java/com/aurx/core/services/impl/ComponentReportServiceImpl.java [hunk 6]
```diff
@@ -47,30 +54,31 @@ protected void modified() {
...
 
   }
+
+  /**
+   * This Method  Fetch the components Resource List
+   * @return =resourceList
+   */
   @Override
   public List<Resource> fetchComponents() {
+    logger.info("===========fetchComponents methode start================");
```
Comment (akashsen-aurx) [inline core/src/main/java/com/aurx/core/services/impl/ComponentReportServiceImpl.java]: remove this symbol

File: core/src/main/java/com/aurx/core/services/impl/ComponentReportServiceImpl.java [hunk 7]
```diff
@@ -47,30 +54,31 @@ protected void modified() {
     fetchTitle();
 
   }
+
+  /**
+   * This Method  Fetch the components Resource List
+   * @return =resourceList
```
Comment (akashsen-aurx) [inline core/src/main/java/com/aurx/core/services/impl/ComponentReportServiceImpl.java:72]: return list of resource

File: core/src/main/java/com/aurx/core/services/impl/ComponentReportServiceImpl.java [hunk 8]
```diff
@@ -47,30 +54,31 @@ protected void modified() {
...
   public List<Resource> fetchComponents() {
+    logger.info("===========fetchComponents methode start================");
     List<Resource> resourceList = new ArrayList<>();
 
       ResourceResolver resourceResolver = null;
       try {
         resourceResolver = getResourceResolver(resourceResolverFactory);
       } catch (LoginException e) {
-        logger.info("this is  Exception : {}",e.getMessage());
+        logger.info("================Exception : {}",e.getMessage());
```
Comment (akashsen-aurx) [inline core/src/main/java/com/aurx/core/services/impl/ComponentReportServiceImpl.java]: which type of exception please define in log

File: core/src/main/java/com/aurx/core/services/impl/ComponentReportServiceImpl.java [hunk 9]
```diff
@@ -47,30 +54,31 @@ protected void modified() {
...
       if(resourceResolver !=null) {
-        QueryBuilder builder = resourceResolver.adaptTo(QueryBuilder.class);
-        SearchResult result = null;
-        if (builder != null) {
-          Query query =
-              builder.createQuery(PredicateGroup.create(predicateMap),
-                  resourceResolver.adaptTo(Session.class));
-          result = query.getResult();
-        }
+        QueryBuilderUtilImpl queryBuilderUtil= new QueryBuilderUtilImpl();
```
Comment (akashsen-aurx) [inline core/src/main/java/com/aurx/core/services/impl/ComponentReportServiceImpl.java]: @Reference se call kro
No need of object creation

File: core/src/main/java/com/aurx/core/servlets/ComponentReportServlet.java [hunk 10]
```diff
@@ -33,7 +38,7 @@
 })
 public class ComponentReportServlet extends SlingAllMethodsServlet {
 
-  private final Logger logger = LoggerFactory.getLogger(ComponentReportServlet.class);
+  private static final  Logger logger = LoggerFactory.getLogger(ComponentReportServlet.class);
```
Comment (akashsen-aurx) [inline core/src/main/java/com/aurx/core/servlets/ComponentReportServlet.java]: java doc

File: core/src/main/java/com/aurx/core/utils/ComponentUtils.java (hunk 1 above)
Comment (akashsen-aurx) [reply core/src/main/java/com/aurx/core/utils/ComponentUtils.java]: name of class will be ApplicationConstant

File: core/src/main/java/com/aurx/core/utils/QueryBuilderUtilImpl.java [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+import java.util.Map;
+import javax.jcr.Session;
+import org.apache.sling.api.resource.ResourceResolver;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class QueryBuilderUtilImpl implements QueryBuilderUtil {
+private static final Logger LOGGER = LoggerFactory.getLogger(QueryBuilderUtilImpl.class);
+  /**
+   *
```
Comment (akashsen-aurx) [inline core/src/main/java/com/aurx/core/utils/QueryBuilderUtilImpl.java:25]: give me explanation of method

File: core/src/main/java/com/aurx/core/utils/QueryBuilderUtilImpl.java [hunk 12]
```diff
@@ -0,0 +1,37 @@
...
+   * @param resourceResolver
+   * @param predicateMap
+   * @return = SearchResult object
+   */
+  @Override
+  public SearchResult getQueryuilderResult(ResourceResolver resourceResolver,
+      Map<String, String> predicateMap) {
+LOGGER.info("========SearchResult method start ========== resourceResolver : {} ,predicateMap :{}",resourceResolver,predicateMap);
```
Comment (akashsen-aurx) [inline core/src/main/java/com/aurx/core/utils/QueryBuilderUtilImpl.java]: resolve logger problem

**Review by akashsen-aurx — APPROVED** [review]

**Review by akashsen-aurx — APPROVED** [review]

File: core/src/main/java/com/aurx/core/models/ComponentReportModel.java [hunk 13]
```diff
@@ -12,17 +12,35 @@
...
+   * The logger - logger  object
+   */
+  private static  final Logger logger= LoggerFactory.getLogger(ComponentReportModel.class);
+
+  /**
+   * The method is used to add
+   */
   @PostConstruct
   protected void init(){
     comopnentHierachyList = new ArrayList<>();
```
Comment (abhinav-aggarwal-aem) [inline core/src/main/java/com/aurx/core/models/ComponentReportModel.java:40]: start method logger

File: core/src/main/java/com/aurx/core/constant/ApplicationConstants.java [hunk 14]
```diff
@@ -0,0 +1,19 @@
+package com.aurx.core.constant;
+
+/**
+ * This class provide the ApplicationConstants
+ */
+public final class ApplicationConstants {
+
+  public static final String SLING_RESOURCE_TYPE = "sling:resourceType";
```
Comment (abhinav-aggarwal-aem) [inline core/src/main/java/com/aurx/core/constant/ApplicationConstants.java:8]: java docs

File: core/src/main/java/com/aurx/core/pojo/ComponentReport.java [hunk 15]
```diff
@@ -2,57 +2,126 @@
...
+  }
 
+  /**
+   * This Constructor used to set the properties
+   * @param title
+   * @param group
+   * @param pathOfComponent
+   */
   public ComponentReport(String title, String group, String pathOfComponent) {
     this.title = title;
```
Comment (abhinav-aggarwal-aem) [inline core/src/main/java/com/aurx/core/pojo/ComponentReport.java:44]: start and end logger

File: core/src/main/java/com/aurx/core/services/QueryBuilderUtil.java [hunk 16]
```diff
@@ -0,0 +1,9 @@
+package com.aurx.core.services;
+
+import com.day.cq.search.result.SearchResult;
+import java.util.Map;
+import org.apache.sling.api.resource.ResourceResolver;
+
+public interface QueryBuilderUtil {
```
Comment (abhinav-aggarwal-aem) [inline core/src/main/java/com/aurx/core/services/QueryBuilderUtil.java:7]: java docs

File: core/src/main/java/com/aurx/core/services/impl/ComponentReportServiceImpl.java [hunk 17]
```diff
@@ -28,80 +31,108 @@
...
+
+  /**
+   * method start when we deploy the bundle in aem so in this method we fetch the title of
+   * components
+   */
   @Activate
+  @Modified
   protected void activate() {
     logger.info("inside the active methode component  service : ");
     fetchTitle();
```
Comment (abhinav-aggarwal-aem) [inline core/src/main/java/com/aurx/core/services/impl/ComponentReportServiceImpl.java:65]: please correct the method name

**Pull request merged**